packages/edit-navigation/src/components/menu-editor/use-navigation-blocks.js

```diff
@@ -87,10 +110,19 @@ export default function useNavigationBlocks( menuId ) {
 			innerBlocks.map( ( block ) => block.clientId )
 		);
 
-		// Disable reason, this code will eventually be implemented.
-		// eslint-disable-next-line no-unused-vars
-		for ( const clientId of deletedClientIds ) {
-			// TODO - delete menu items.
+		if ( deletedClientIds.length > 0 ) {
+			deleteMenuItems( deletedClientIds, menuItemsRef.current );
+
+			createSuccessNotice(
+				sprintf(
+					/* translators: %s: block pattern title. */
+					__( 'Deleted %s menu items.' ),
+					deletedClientIds.length
+				),
+				{
+					type: 'snackbar',
+				}
+			);
```

Comment:
Should we only have a single notification in place for all the operations? I imagine seeing "Created 10 menu items", "Deleted 2 menu items" and "Updated 3 menu items" would be pretty confusing.

Reply:
Sure!